pkgs/development/libraries/arrow-cpp/default.nix

```diff
@@ -32,28 +48,28 @@ in stdenv.mkDerivation rec {
     sha256 = "1xl7z0vwbn5iycg7amka9jd6hxd8nmfk7nahi4p9w2bnw9f0wcrl";
   };
 
+  ARROW_MIMALLOC_URL = fetchurl {
+    # From
+    # ./cpp/cmake_modules/ThirdpartyToolchain.cmake
+    # ./cpp/thirdparty/versions.txt
+    url =
+      "https://github.com/microsoft/mimalloc/archive/v1.6.3.tar.gz";
+    sha256 = "0pia8b4acv1w8qzcpc9i1a2fasnn3rmp996k0l87p2di0lbls0w5";
+  };
+
   patches = [
     # patch to fix python-test
     ./darwin.patch
-
-    # fix musl build
-    (fetchpatch {
-      url = "https://github.com/apache/arrow/commit/de4168786dfd8ab932f48801e0a7a6b8a370c19d.diff";
-      sha256 = "1nl4y1rwdl0gn67v7l05ibc4lwkn6x7fhwbmslmm08cqmwfjsx3y";
-      stripLen = 1;
-    })
-
-    # fix build for "ZSTD_SOURCE=SYSTEM"
-    (fetchpatch {
-      url = "https://github.com/apache/arrow/commit/13cb3dbded1928d2e96574895bebaf9098a4796d.diff";
-      sha256 = "12z3ys47qp2x8f63lggiyj4xs2kmg804ri4xqysw5krbjz2hr6rb";
-      stripLen = 1;
-    })
-  ] ++ lib.optionals (!enableShared) [
-    # The shared jemalloc lib is unused and breaks in static mode due to missing -fpic.
-    ./jemalloc-disable-shared.patch
   ];
 
+  # Properly exported static targets. Remove at the next version bump.
+  postPatch = ''
+    pushd ..
+    patch -p1 < ${static-utf8proc-patch}
+    patch -p1 < ${static-targets-patch}
```

Comment:
I believe there is no reason to not use `patches` attribute. The `stripLen` should help to achieve that.

Reply:
As far as my knowledge on this goes I'd have to remove the `/cpp` part from `sourceRoot` and cd to the subdirectory in `postPatch`. Maybe there is a better option that I'm missing?

Reply:
There is an option to prepend a string to path. `extraPath`, I think it was

Reply:
I just tried
```
    stripLen = 1;
    extraPrefix = "../";
```
But that fails. `patch` does not seem to like the parent-ref:
```
...
can't find file to patch at input line 72
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- a/../python/manylinux1/build_arrow.sh
|+++ b/../python/manylinux1/build_arrow.sh
--------------------------
...
```

Reply:
I believe the "cpp/" prefix needs to be stripped and “python/“ paths need to be filtered (there is an option for that, probably called `exclude`).

Reply:
Found a way with explicit `includes`, but didn't get to test the static build yet.

Reply:
Had to make one more adjustment, both `pkgsStatic.arrow-cpp` and `python38Packages.pyarrow` worked for my testcases.